zhaquirks/tuya/tuya_trv.py

```diff
@@ -68,6 +68,25 @@ def __init__(self, *args, **kwargs):
         self.add_unsupported_attribute(Thermostat.AttributeDefs.pi_heating_demand.id)
 
 
+class TuyaThermostatV2NoSchedule(TuyaThermostatV2):
+    """Ensures schedule is disabled on system_mode change."""
+
+    async def write_attributes(self, attributes, manufacturer=None):
+        """Defer attributes writing to the set_data tuya command."""
+
+        if (
+            Thermostat.AttributeDefs.system_mode.id in attributes
+            or Thermostat.AttributeDefs.system_mode.name in attributes
+        ):
+            await super().write_attributes(attributes, manufacturer)
+
+            tuya_cluster = self.endpoint.tuya_manufacturer
+            return await tuya_cluster.write_attributes({"schedule_enable": False})
```

Comment:
Hmm, we return the results of the tuya cluster `write_attributest` call here.
We should instead return the results of the actual call on this cluster. This might work:
```suggestion
            results = await super().write_attributes(attributes, manufacturer)
            tuya_cluster = self.endpoint.tuya_manufacturer
            await tuya_cluster.write_attributes({"schedule_enable": False})
            return results
```

Reply:
Actually, we could probably just put `results = await super().write_attributes(attributes, manufacturer)` at the beginning of the method. Then go into the if condition and at the end, just `return results` outside the if block.
Then, we don't need the `else`.